src/mod_ping.erl

```diff
@@ -232,9 +224,37 @@ user_keep_alive(Acc, JID) ->
     start_ping(JID#jid.lserver, JID),
     Acc.
 
+user_ping_timeout(Acc, _JID) -> % TODO is this hook really needed?
+    Acc.
+
+-spec user_ping_response(Acc :: mongoose_acc:t(),
+                         JID :: jid:jid(),
+                         Response :: timeout | jlib:iq(),
+                         TDelta :: pos_integer()) -> mongoose_acc:t().
+user_ping_response(Acc, #jid{server = Server}, Response, TDelta) ->
+    case Response of
+        timeout -> mongoose_metrics:update(Server, [mod_ping, ping_response_timeout], 1);
+        _ -> mongoose_metrics:update(Server, [mod_ping, ping_response], 1)
+    end,
+    mongoose_metrics:update(Server, [mod_ping, ping_response_time], TDelta),
```

Comment:
I think this histogram shouldn't be updated with timeout value. We know what timeout is configured and it makes this histogram useless. For example, median or average may be seriously skewed.